returnn/datasets/util/vocabulary.py

```diff
@@ -162,12 +163,43 @@ def get_seq_indices(self, seq):
 
   def get_seq_labels(self, seq):
     """
-    :param list[int] seq:
+    :param list[int]|int|numpy.ndarray seq: 0D or 1D
     :rtype: str
     """
+    if isinstance(seq, int):
+      seq = [seq]
+    elif isinstance(seq, numpy.ndarray) and seq.ndim == 0:
+      seq = numpy.expand_dims(seq, axis=0)
     return " ".join(map(self.labels.__getitem__, seq))
 
 
+def get_seq_labels_from_labels(seq, labels):
+  """
+  Warning: Deprecated, use the corresponding vocabulary to serialize the data
+
+  :param list[int]|int|numpy.ndarray seq: 0D or 1D
+  :param list[str] labels:
+  :rtype: str
+  """
+  if isinstance(seq, int):
+    seq = [seq]
+  elif isinstance(seq, numpy.ndarray) and seq.ndim == 0:
+    seq = numpy.expand_dims(seq, axis=0)
+  if len(labels) < 1000 and all([len(label) == 1 for label in labels]):
+    # are these actually raw bytes? -> assume utf8
+    if all([ord(label) <= 255 for label in labels]):
+      try:
+        if PY3:
+          return bytes([ord(labels[c]) for c in seq]).decode("utf8")
+        else:
+          return b"".join([bytes(labels[c]) for c in seq]).decode("utf8")
+      except UnicodeDecodeError:
+        pass  # pass on to default case
+    return "".join(map(labels.__getitem__, seq))
+  else:
+    return " ".join(map(labels.__getitem__, seq))
```

Comment:
This logic should be removed, and you should instead wrap it up in a `Vocabulary` and then call to `get_seq_labels`. Maybe it could use a similar heuristic on what type of vocab class you use here.

Reply:
Maybe you should make a separate function `create_vocab_from_labels`.